src/Http/Http.Extensions/src/HttpRequestJsonExtensions.cs

```diff
@@ -32,13 +32,14 @@ public static class HttpRequestJsonExtensions
     /// <param name="request">The request to read from.</param>
     /// <param name="cancellationToken">A <see cref="CancellationToken"/> used to cancel the operation.</param>
     /// <returns>The task object representing the asynchronous operation.</returns>
-    [RequiresUnreferencedCode(RequiresUnreferencedCodeMessage)]
```

Comment:
Why havne't we done this for all methods in this class? For example, 

```C#
    public static async ValueTask<TValue?> ReadFromJsonAsync<TValue>(
        this HttpRequest request,
        JsonSerializerOptions? options,
        CancellationToken cancellationToken = default)
```

Reply:
Similar question for `HttpResponseJsonExtensions`.

Reply:
When the user provide a `JsonSerializerOptions` the `TypeInfoResolver` could be null, default value, and we cannot call `GetTypeInfo` https://github.com/dotnet/runtime/issues/80529